File: microservers/immobilierservice/src/main/java/io/multi/immobilierservice/dto/BrouillonSaveRequest.java
package io.multi.immobilierservice.dto;

import jakarta.validation.constraints.Max;
import jakarta.validation.constraints.Min;
import jakarta.validation.constraints.NotNull;
import lombok.Data;

import java.util.Map;

@Data
public class BrouillonSaveRequest {

    @NotNull
    @Min(1) @Max(4)
    private Integer etapeActuelle;

    /**
     * Structure libre du formulaire (steps Bien/Lieu/Prix/Photos).
     * Le backend ne valide pas le contenu — c'est juste persisté en JSONB.
     */
    @NotNull
    private Map<String, Object> donneesJson;
}

File: mobile_app/lib/features/immo/models/brouillon_save_request.dart
/// DTO de requête pour `POST /immo/brouillons` et `PUT /immo/brouillons/{uuid}`.
///
/// Conforme à `BrouillonSaveRequest.java` côté backend :
/// - etapeActuelle : 1..4 (NotNull, Min 1 Max 4)
/// - donneesJson : Map libre (NotNull, mais peut être {})
class BrouillonSaveRequest {
  final int etapeActuelle;
  final Map<String, dynamic> donneesJson;

  const BrouillonSaveRequest({
    required this.etapeActuelle,
    required this.donneesJson,
  });

  Map<String, dynamic> toJson() => {
        'etapeActuelle': etapeActuelle,
        'donneesJson': donneesJson,
      };
}
